### DISCUSSION TITLE
web3modal in iFrame

### DISCUSSION THREAD

**Discussion by JacopoNeroni** [WalletConnect/walletconnect-monorepo#2433] [Developer Support]
I have integrated a web app using web3modal v2 (https://swappy.exchange/trade?refcode=0000001) on https://pepeswap.trade/ via an iframe on a different domain than the one where the web app resides.
The problem is that if launched from a browser on a mobile device (safari, chrome etc), it does not open the wallet to connect, no event happens. Obviously if I use the link directly without the encapsulation in the iframe, it works fine, It works even if I open it from the dapp browser. 
I think it is a problem related to connection injection. Do you have any tips or any other methods to be able to get the same result?

**Comment by GiuBlockchainDEV** [comment]
Up

**Comment by glitch-txs** [comment]
Which phone are you having issues with? Tested with Android and was able to connect

**Reply by JacopoNeroni** [reply]
iphone ios, on android works fine

**Comment by finessevanes** [comment]
@JacopoNeroni can you include a screenshot/video of the issue?

**Reply by JacopoNeroni** [reply]
https://github.com/WalletConnect/walletconnect-monorepo/assets/95048349/6d55cb9d-5281-454b-8c15-3122f41b2f44

**Reply by JacopoNeroni** [reply]
Obviously if I do it directly from https://swappy.exchange/trade?refcode=0000001 without going through the iFrame it works perfectly. The problem is only on iOS, not Android.

**Reply by JacopoNeroni** [reply]
up

**Reply by finessevanes** [reply]
@JacopoNeroni can you share the SDK and version?

https://github.com/WalletConnect/walletconnect-monorepo/assets/44731696/7427c600-8df6-4793-8eda-54e1bd67463e

**Reply by JacopoNeroni** [reply]
"@web3modal/ethereum": "^2.2.0"
 "@web3modal/react": "^2.2.0"

but the problem is only on iOS not in Android, your phone is Android. It also works in ios without the iFrame on swappy.exchange. If you open it in www.pepeswap.trade it does not work only on iOS since it is embedded in an iFrame.
